packages/telemetry/src/slog-to-otel.js

```diff
@@ -104,13 +110,124 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
   const dbTransactionManager = makeKVDatabaseTransactionManager(db);
   const kernelPromiseToSendingCause = makeKVStringStore('kernelPromise', db);
 
-  const extractMethod = methargs => JSON.parse(methargs.body)[0];
+  /** @type {WeakMap<Promise | object, {kref: string, iface: string | undefined}>} */
+  const presenceMap = new WeakMap();
+  /** @type {Map<string, Promise | object>} */
+  const valCache = new Map();
+
+  const slotToVal = (kref, iface) => {
+    let val = valCache.get(kref);
+    if (val) {
+      return val;
+    }
+
+    if (iface) {
+      if (!iface.startsWith('Alleged: ') && iface !== 'Remotable') {
+        iface = `Alleged: ${iface}`;
+      }
+    }
+
+    kref[0] === 'k' || assert.fail(`Unexpected non-kernel ref ${kref}`);
+    val = harden(
+      kref.startsWith('kp')
+        ? new Promise(sink)
+        : Remotable(iface || 'Remotable'),
+    );
+    valCache.set(kref, val);
+    presenceMap.set(val, { kref, iface });
+
+    return val;
```

Comment:
Why not just:

```suggestion
  if (!valCache.has(kref)) {
    valCache.set(kref, { kref, iface });
  }
  return valCache.get(kref);
```

Reply:
As discussed, marshal now enforces that a remotable / promise actually checks against `passStyleOf`, so `Remotable` / `Promise` is necessary.

The `presenceMap` however is a leftover from a large change where I used serialization. I can remove.